openpgp/packet/aead_crypter.go

```diff
@@ -62,68 +60,54 @@ func (wo *aeadCrypter) incrementIndex() error {
 type aeadDecrypter struct {
 	aeadCrypter           // Embedded ciphertext opener
 	reader      io.Reader // 'reader' is a partialLengthReader
+	chunkBytes  []byte
 	peekedBytes []byte    // Used to detect last chunk
-	eof         bool
+	buffer      []byte    // Buffered decrypted bytes
 }
 
 // Read decrypts bytes and reads them into dst. It decrypts when necessary and
 // buffers extra decrypted bytes. It returns the number of bytes copied into dst
 // and an error.
 func (ar *aeadDecrypter) Read(dst []byte) (n int, err error) {
 	// Return buffered plaintext bytes from previous calls
-	if ar.buffer.Len() > 0 {
-		return ar.buffer.Read(dst)
-	}
-
-	// Return EOF if we've previously validated the final tag
-	if ar.eof {
-		return 0, io.EOF
+	if len(ar.buffer) > 0 {
+		n = copy(dst, ar.buffer)
+		ar.buffer = ar.buffer[n:]
+		return
 	}
 
 	// Read a chunk
 	tagLen := ar.aead.Overhead()
-	cipherChunkBuf := new(bytes.Buffer)
-	_, errRead := io.CopyN(cipherChunkBuf, ar.reader, int64(ar.chunkSize+tagLen))
-	cipherChunk := cipherChunkBuf.Bytes()
-	if errRead != nil && errRead != io.EOF {
+	copy(ar.chunkBytes, ar.peekedBytes)
```

Comment:
`peekedBytes` is filled the first time [here](https://github.com/ProtonMail/go-crypto/blob/172524b3901a903b3b4d3b23c5d86023fd665db3/openpgp/packet/symmetrically_encrypted_aead.go#L77) and [here](https://github.com/ProtonMail/go-crypto/blob/172524b3901a903b3b4d3b23c5d86023fd665db3/openpgp/packet/aead_encrypted.go#L72).

Reply:
Ah ok, thanks for the pointer. It had to be since the tests passed, Maybe we should add a comment. 
`//  peekedBytes is pre-filled before calling this function` or something like that.